File: frontend/src/types/telemetry.ts
export interface TelemetryReading {
  timestamp: string;
  speedKph: number;
  rpm: number;
  engineTempC: number;
  batteryVoltage: number;
  gear: string;
  lat: number;
  lng: number;
}

export type FaultSeverity = 'warn' | 'critical';

export interface FaultAlert {
  id: string;
  code: string;
  message: string;
  severity: FaultSeverity;
  triggeredAt: string;
  active: boolean;
}

export interface TripSummary {
  tripId: string;
  distanceKm: number;
  avgSpeedKph: number;
  idleMinutes: number;
  fuelUsedLiters: number;
  startedAt: string;
  endedAt: string;
}

export interface HistoricalPoint {
  timestamp: string;
  engineTempC: number;
  batteryVoltage: number;
  rpm: number;
}

export type ConnectionStatus = 'connecting' | 'live' | 'offline';

File: backend/app/schemas/telemetry.py
from datetime import datetime

from pydantic import BaseModel, Field


class TelemetryReadingCreate(BaseModel):
    speedKph: float = Field(..., alias="speedKph")
    rpm: int
    engineTempC: float = Field(..., alias="engineTempC")
    batteryVoltage: float = Field(..., alias="batteryVoltage")
    gear: str
    lat: float
    lng: float
    vehicleId: str = Field(default="vehicle-001", alias="vehicleId")
    timestamp: datetime | None = None

    model_config = {"populate_by_name": True}


class TelemetryReadingResponse(BaseModel):
    id: str
    timestamp: datetime
    speedKph: float
    rpm: int
    engineTempC: float
    batteryVoltage: float
    gear: str
    lat: float
    lng: float

    model_config = {"from_attributes": True}


class HistoricalPoint(BaseModel):
    timestamp: datetime
    engineTempC: float
    batteryVoltage: float
    rpm: int
